smartlearner/utils.py

```diff
@@ -23,3 +25,73 @@ def save_dict_to_json_file(path, dictionary):
 def load_dict_from_json_file(path):
     with open(path, "r") as json_file:
         return json.loads(json_file.read())
+
+
+def split_dataset(dataset, proportions):
+    """ Split a dataset into many smaller datasets.
+
+    Parameters
+    ----------
+    dataset : Dataset
+    proportions : [numbers]
+        A list of numbers giving the proportions of the dataset to be included in each
+        sub-datasets. The proportions are computed according to the sum all the numbers.
+
+    Returns
+    -------
+    [Datasets]
+        A list of datasets which respect the given `proportions`.
+
+    """
+    from .interfaces.dataset import Dataset
+    indices = np.cumsum(np.ceil(np.array(proportions) / np.sum(proportions) * len(dataset)))
+    indices[-1] = len(dataset)
+    dsets = []
+
+    for s, f in zip([0] + list(indices), indices):
+        s, f = int(s), int(f)
+
+        covars = dataset.inputs.get_value()[s:f]
+        targets = dataset.targets.get_value()[s:f] if dataset.has_targets else None
+
+        dset = Dataset(covars, targets, dataset.name + '_' + str(s) + 'to' + str(f))
+        dset.symb_inputs = dataset.symb_inputs
+        dset.symb_targets = dataset.symb_targets
+
+        dsets.append(dset)
+
+    return dsets
+
+
+def kfold(dataset, k):
+    return split_dataset(dataset, np.ones((k,)))
+
+
+def growing_sequential_kfold(dataset, k):
+    """ Builds `k` growing validations triplets.
+
+    Parameters
+    ----------
+    dataset : Dataset
+    k : int
+        The number of triplets required as validations sets.
+
+    Returns
+    -------
+    [[Dataset]]
+        Returns a list of list of three datasets (training set, validation set, test set) where the length of the
+        training set is growing for each further triplets.
+    """
+    if k < 2:
+        raise ValueError("Parameter k has to be greater than 1.")
+    folds = []
+
+    for i in range(1, k + 1):
+        if not k - i:
```

Comment:
No need to check if `k - i == 0`. Calling `folds.append(split_dataset(dataset, [i, 1, 1, k - i])[:3])` directly is valid. If `k-i==0` then the last dset returned is a valid `Dataset` object with length of 0 :)